src/main/java/seedu/module/model/ModuleBook.java

```diff
@@ -78,7 +79,14 @@ public boolean hasTask(Task task) {
      * The task must not already exist in the module book.
      */
     public void addTask(Task p) {
-        tasks.add(p);
+        assert(p != null);
+        if (ModuleManager.getListOfExistingModules().contains(p.getModule().toString())) {
```

Comment:
Will it be bettter to implement a method in ModuleManager to check this? Since the implementation here may violate the Law of Demeter.

Reply:
Yes i agree with @QY-H00 here. Was going to mention that highly likely law of demeter is violated here. Although, if u think the violation is justified, its fine (according to prof, can justfiy for convenience sake at times). If not, i think its better to add a method in!

Reply:
Agree! Will resolve now